packages/suite/src/views/firmware/index.tsx

```
@@ -180,406 +41,167 @@ type Props = ReturnType<typeof mapDispatchToProps> &
+                    Heading: <InitialStep.Heading />,
+                    Body: <InitialStep.Body />,
+                    BottomBar: <InitialStep.BottomBar />,
+                };
+            case 'check-seed':
+                return {
+                    Heading: <CommonHeading />,
+                    Body: <CheckSeedStep.Body />,
+                    BottomBar: <CheckSeedStep.BottomBar />,
+                };
+            case 'waiting-for-bootloader':
+                return {
+                    Heading: <CommonHeading />,
+                    Body: <ReconnectInBootloaderStep.Body />,
+                    BottomBar: <ReconnectInBootloaderStep.BottomBar />,
+                };
+            case 'waiting-for-confirmation':
+            case 'installing':
+            case 'started':
+            case 'downloading':
+            case 'check-fingerprint':
+            case 'wait-for-reboot':
+            case 'unplug':
+                return {
+                    Heading: <CommonHeading />,
+                    Body: <FirmwareProgressStep.Body />,
+                    BottomBar: null,
+                };
+            case 'reconnect-in-normal':
+                return {
+                    Heading: <CommonHeading />,
+                    Body: <ReconnectInNormalStep.Body />,
+                    BottomBar: <ReconnectInNormalStep.BottomBar />,
+                };
+            case 'partially-done':
+                return {
+                    Heading: <CommonHeading />,
+                    Body: <PartiallyDoneStep.Body />,
+                    BottomBar: <ContinueButton onClick={resetReducer} />,
+                };
+            case 'done':
+                return {
+                    Heading: <DoneStep.Heading />,
+                    Body: <DoneStep.Body />,
+                    BottomBar: <CloseButton onClick={onClose} />,
+                };
+
+            default:
+                // should never get here
+                return null;
```

If we don't ever expect it to get there, we should throw an error. Just in case it happens, we can log it.

---

probably reasonable